lib/pipeline.js

```diff
@@ -701,15 +703,26 @@ class PipelineModel extends BaseModel {
             // eslint-disable-next-line no-await-in-loop
             const permission = await user.getPermissions(this.scmUri);
 
-            if (permission.push) {
-                return user;
+            if (!permission.push) {
+                delete newAdmins[username];
```

Comment:
All admins who doesn't have permission will removed here. The [reference of executor queue](https://github.com/screwdriver-cd/executor-queue/pull/28/files) is no problem because all users in admins table have proper permission.